tree/dataframe/inc/ROOT/RResultPtr.hxx

```diff
@@ -158,14 +159,15 @@ public:
    RResultPtr &operator=(const RResultPtr &) = default;
    RResultPtr &operator=(RResultPtr &&) = default;
    explicit operator bool() const { return bool(fObjPtr); }
-   template <typename TO, typename std::enable_if<std::is_convertible<T, TO>::value, int>::type = 0>
-   operator RResultPtr<TO>() const
+
+   /// Convert a RResultPtr<T2> to a RResultPtr<T>.
+   ///
+   /// Useful e.g. to store a number of RResultPtr<TH1D> and RResultPtr<TH2D> in a std::vector<RResultPtr<TH1>>.
+   /// The requirements on T2 and T are the same as for conversion between std::shared_ptr<T2> and std::shared_ptr<T>.
+   template <typename T2, typename std::enable_if<std::is_constructible<std::shared_ptr<T>, std::shared_ptr<T2>>::value,
+                                                  int>::type = 0>
```

Comment:
```suggestion
   template <typename T2, decltype(std::shared_ptr<T>(std::declval<T2*>()))* = nullptr>
```

Reply:
Should be `decltype(std::shared_ptr<T>(std::declval<std::shared_ptr<T2>>()))* = nullptr` to be precise, right?

Do you find this more readable? I have a slight preference for the "plain english" `enable_if<is_constructible<...>>`

Reply:
That's totally fine if you prefer the spelled-out version, go for it!